code/modules/mob/mob.dm

```diff
@@ -1773,5 +1777,48 @@ mob/proc/on_foot()
 	ear_deaf = init_deaf
 	clear_fullscreen("blind")
 
+/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE)
+	if(!target_type)
+		if(transmogged_from)
+			transmogged_from.forceMove(loc)
+			if(key)
+				transmogged_from.key = key
+			transmogged_from.timestopped = 0
+			if(istype(transmogged_from, /mob/living/carbon))
+				var/mob/living/carbon/C = transmogged_from
+				if(C.wear_mask && istype(C.wear_mask, /obj/item/clothing/mask/morphing))
```

Comment:
Eeeeh. Really not keen on the hardcoded typepath.

Then again it's probably fine due to how specific of a use case this is.

Also `get_item_by_slot`